packages/main/src/files/file-manager.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { RequestOptions } from "../../types";
+import { readFileSync } from "fs";
+import { FilesRequestUrl, getHeaders, makeFilesRequest } from "./request";
+import {
+  FileMetadata,
+  FileMetadataResponse,
+  ListFilesResponse,
+  ListParams,
+  UploadFileResponse,
+} from "./types";
+import { FilesTask } from "./constants";
+
+// Internal type, metadata sent in the upload
+export interface UploadMetadata {
+  name?: string;
+  ["display_name"]?: string;
+}
+
+/**
+ * Class for managing GoogleAI file uploads.
+ * @public
+ */
+export class GoogleAIFileManager {
+  constructor(
+    public apiKey: string,
+    private _requestOptions?: RequestOptions,
+  ) {}
+
+  /**
+   * Upload a file
+   */
+  async uploadFile(
+    filePath: string,
+    fileMetadata: FileMetadata,
+  ): Promise<UploadFileResponse> {
+    const file = readFileSync(filePath);
+    const url = new FilesRequestUrl(
+      FilesTask.UPLOAD,
+      this.apiKey,
+      this._requestOptions,
+    );
+
+    const uploadHeaders = getHeaders(url);
+    const boundary = genBoundary();
+    uploadHeaders.append("X-Goog-Upload-Protocol", "multipart");
+    uploadHeaders.append(
+      "Content-Type",
+      `multipart/related; boundary=${boundary}`,
+    );
+
+    const uploadMetadata: FileMetadata = {
+      mimeType: fileMetadata.mimeType,
+    };
+    uploadMetadata.displayName = fileMetadata.displayName;
+    uploadMetadata.name = fileMetadata.name;
```

Comment:
nit: Do we want to set the `name` and the `displayName` at initialization?
```
const uploadMetadata: FileMetadata = {
      mimeType: fileMetadata.mimeType,
      displayName:...
    };
```

Reply:
Done, that is a bit cleaner.

Reply:
Thanks! We can now remove line 76, 77. and 78-80, right?

Reply:
Done